arroyo/processing/strategies/dead_letter_queue/policies/produce.py

```diff
@@ -0,0 +1,85 @@
+import json
+import time
+from collections import deque
+from concurrent.futures import Future
+from datetime import datetime
+from typing import Any, Deque, Mapping, Optional
+
+from arroyo.backends.abstract import Producer
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessage,
+    InvalidMessages,
+    JSONSerializable,
+)
+from arroyo.types import Message, Topic
+from arroyo.utils.codecs import Encoder
+from arroyo.utils.metrics import get_metrics
+
+MAX_QUEUE_SIZE = 5000
+DATE_TIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+class JSONMessageEncoder(Encoder[bytes, Mapping[str, JSONSerializable]]):
+    def __default(self, value: Any) -> str:
+        if isinstance(value, datetime):
+            return value.strftime(DATE_TIME_FORMAT)
+        else:
+            raise TypeError
```

Comment:
Do you need this?

Reply:
Yeah I think it needs a default "return" and I don't think it should arbitrarily return an empty string or something in the else case. That will likely never be raised but it's just there for clarity?